app/assets/javascripts/discourse/app/lib/utilities.js

```diff
@@ -349,6 +349,15 @@ const toArray = (items) => {
   return items;
 };
 
+const gifInDisguise = (clipboard) => {
+  return (
+    clipboard.files.length === 1 &&
+    clipboard.files[0].type === "image/png" &&
```

Comment:
No need to change this, but unless you actually want to be sure there are not 2 files here, you could just have done:

```
clipboard.files[0]?.type === "image/png"
```

Reply:
Oh but I want to be sure that there are no two files indeed :sweat_smile: 

It's a weird heuristic to detect when we have a gif in disguise that becase a PNG when someone right cliked a gif and selected "Copy image" in a browser. If there are two files, we can fail the detection because it's not the case!